parser/compiler.go

```diff
@@ -328,18 +329,104 @@ type entry struct {
 	*symbol
 }
 
+func (e entry) String() string {
+	if e.symbol != nil {
+		return fmt.Sprintf("name: %s,local: %t, i: %d, k: %d, t: %s, v: %v",
+			e.name,
+			e.symbol.local,
+			e.symbol.index,
+			e.symbol.kind,
+			e.symbol.Type,
+			e.symbol.value,
+		)
+	}
+
+	return e.name
+}
+
 func (c *Compiler) PrintData() {
+	dict := c.symbolsByIndex()
+
+	fmt.Fprintln(os.Stderr, "# Data:")
+	for i, d := range c.Data {
+		fmt.Fprintf(os.Stderr, "%4d %T %v %v\n", i, d.Data.Interface(), d.Data, dict[i])
+	}
+}
+
+func (c *Compiler) symbolsByIndex() map[int]entry {
 	dict := map[int]entry{}
 	for name, sym := range c.symbols {
-		if !sym.used || sym.local || sym.kind == symLabel {
+		if sym.index == unsetAddr {
 			continue
 		}
 		dict[sym.index] = entry{name, sym}
 	}
-	fmt.Fprintln(os.Stderr, "# Data:")
+
+	return dict
+}
+
```

Comment:
Could you explain the motivation of this feature in a comment, specially the need for portability and independence of VM implementation?

Reply:
Done! [`7a9ac73` (#7)](https://github.com/mvertes/parscan/pull/7/commits/7a9ac73037f207e3895332e5ba2b30d465c9c339)